bofire/strategies/doe/objective.py

```diff
@@ -204,6 +203,15 @@ def __init__(
                 If None is provided, the default options (maxiter = 500) are used.
         """
 
+        try:
+            from cyipopt import minimize_ipopt  # type: ignore
+        except ImportError as e:
+            warnings.warn(e.msg)
+            warnings.warn(
+                "please run `conda install -c conda-forge cyipopt` for this functionality.",
```

Comment:
I would not put out warnings for missing libraries Because I think more often then not it is intended that library is missing. And if someone really wants to use something they have not installed, they will get an error anyway.